server/registry/internal/storage/gorm/client.go

```diff
@@ -349,3 +353,38 @@ func (c *Client) GetRecentDeploymentRevisions(ctx context.Context, offset int32,
 func (c *Client) Migrate(kind string) error {
 	return c.db.AutoMigrate(entities...)
 }
+
+func (c *Client) GetStorage() (*rpc.Storage, error) {
```

Comment:
Consider splitting this into separate methods, e.g. `DatabaseName() (string, error)`, `TableNames() ([]string, error)`, and `RowCount(table string) int`. The RPC handler can use these to build a response, consistent with how we implement other methods.

Each individual method is easier to understand than one larger and more vaguely named `GetStorage()` method, and it decrease coupling between the storage implementation and our API by keeping RPCs out of the storage package.

On the other hand, I can see value in your current approach where the storage layer responds with RPC messages. If we consistently did that, the storage models would become internal to the storage layer and the RPCs would no longer have to manage that type conversion.

Reply:
Good points - I've restructured this to use the methods that you suggested and build the response in the action handler. I'm not sure that we need to keep the proto models out of the storage layer, but it doesn't seem worthwhile to bring them in for a small feature like this.